pkg/cache/application.go

```diff
@@ -507,6 +508,19 @@ func (app *Application) handleCompleteApplicationEvent(event *fsm.Event) {
 	// TODO app lifecycle updates
 }
 
+func (app *Application) handleFailApplicationEvent(event *fsm.Event) {
+	// unallocated task states include New, Pending and Scheduling
+	unalloc := app.getTasks(events.States().Task.New)
+	unalloc = append(unalloc, app.getTasks(events.States().Task.Pending)...)
+	unalloc = append(unalloc, app.getTasks(events.States().Task.Scheduling)...)
+	// publish pod level event to unallocated pods
+	for _, task := range unalloc {
+		events.GetRecorder().Eventf(task.GetTaskPod(), v1.EventTypeWarning, "ApplicationFailed",
```

Comment:
We need to print the message about this failed app, otherwise, users will be confused about what happened.
This can get via `si.UpdatedApplication.Message`, so in the `scheduler_callback.go`, we need to add that in the FailApplication event as well.

Reply:
OK, done